northstar/src/runtime/state.rs

```diff
@@ -240,12 +240,29 @@ impl State {
             .collect::<Vec<_>>();
 
         // Add resources of containers that have the autostart flag set
-        for (container, _) in &autostarts {
+        for (container, autostart) in &autostarts {
             let manifest = self.manifest(container)?;
             for mount in manifest.mounts.values() {
                 if let Mount::Resource(Resource { name, version, .. }) = mount {
-                    let container = Container::new(name.clone(), version.clone());
-                    to_mount.push(container);
+                    if let Some(resource) =
+                        find_resource(name, version, &mut self.containers.keys().cloned())
+                    {
+                        to_mount.push(resource.clone());
+                    } else {
+                        match autostart {
+                            Autostart::Relaxed => {
+                                warn!("failed to autostart relaxed {}: missing resource {} version {}", container, name, version.to_string());
+                            }
+                            Autostart::Critical => {
+                                error!("failed to autostart critical {}: missing resource {} version {}", container, name, version.to_string());
+                                return Err(Error::StartContainerMissingResource(
+                                    container.clone(),
+                                    name.clone(),
+                                    version.to_string(),
+                                ));
+                            }
+                        }
+                    }
```

Comment:
This block is hard to read because of the high level of indentation. Can we simplify it?

Reply:
I extracted the autostart error handling into its own function.